app/scripts/controllers/serviceInstance.js

```diff
@@ -42,28 +65,76 @@ angular.module('openshiftConsole')
+        $scope.parameterData[key] = $scope.parameterSchema.properties[key].default;
+      });
+
+      $scope.parameterData = angular.extend($scope.parameterData, _.get($scope.serviceInstance, 'spec.parameters', {}));
+
+      if (AuthorizationService.canI('secrets', 'get', $scope.projectName)) {
+        _.each(_.get($scope.serviceInstance, 'spec.parametersFrom'), function (parametersSource) {
+          secretWatchers.push(DataService.watchObject("secrets", _.get(parametersSource, 'secretKeyRef.name'), $scope.projectContext, function (secret) {
+            try {
+              _.extend($scope.parameterData, JSON.parse(SecretsService.decodeSecretData(secret.data)[parametersSource.secretKeyRef.key]));
+            } catch (e) {
+              this.Logger.warn('Unable to load parameters from secret ' + _.get(parametersSource, 'secretKeyRef.name'), e);
+            }
+          }));
+        });
+      }
+    };
+
+    var updateEditable = function() {
+      if (!$scope.plan || !$scope.serviceClass || !$scope.serviceInstance) {
+        return;
+      }
+
+      var updateSchema = _.get($scope.plan, 'spec.instanceUpdateParameterSchema');
+      var planUpdatable = (_.size(_.get(updateSchema, 'properties')) > 0) || (_.get($scope.serviceClass, 'spec.planUpdatable') && (_.size($scope.servicePlans) > 1));
+
+      $scope.editAvailable = planUpdatable && serviceInstanceReady($scope.serviceInstance) && !_.get($scope.serviceInstance, 'metadata.deletionTimestamp');
+    };
+
+    var updateParameterSchema = function() {
+      $scope.parameterFormDefinition = angular.copy(_.get($scope.plan, 'spec.externalMetadata.schemas.service_instance.update.openshift_form_definition'));
+      $scope.parameterSchema = _.get($scope.plan, 'spec.instanceCreateParameterSchema');
+    };
+
     var updateServiceClass = function() {
       // If we've previously loaded the service class or a request is in flight, don't do anything.
-      if ($scope.serviceClass || serviceClassPromise) {
+      if (!$scope.serviceInstance || $scope.serviceClass || serviceClassPromise) {
         return;
       }
 
-      serviceClassPromise = ServiceInstancesService.fetchServiceClassForInstance($scope.serviceInstance).then(function(serviceClass) {
+      serviceClassPromise = ServiceInstancesService.fetchServiceClassForInstance($scope.serviceInstance).then(function (serviceClass) {
         $scope.serviceClass = serviceClass;
-        $scope.displayName = serviceInstanceDisplayName($scope.serviceInstance, serviceClass);
+        $scope.displayName = serviceInstanceDisplayName($scope.serviceInstance, $scope.serviceClass);
+
         updateBreadcrumbs();
         serviceClassPromise = null;
-      });
-    };
 
-    var updatePlan = function() {
-      if (ServiceInstancesService.isCurrentPlan($scope.serviceInstance, $scope.plan)) {
-        return;
-      }
+        Catalog.getServicePlans().then(function (plans) {
```

Comment:
When we fix the removed plans, let's use a field filter on list to get just the plans for this service class and filter removed plans client side (follow on)